pinot-druid/src/main/java1/org/apache/pinot/druid/data/readers/DruidSegmentRecordReader.java

```diff
@@ -0,0 +1,183 @@
+
+    try {
+      QueryableIndex index = indexIO.loadIndex(new File(indexPath));
+      LOGGER.info("Creating segment from path: {}", indexPath);
+      QueryableIndexStorageAdapter adapter = new QueryableIndexStorageAdapter(index);
+
+      // A Sequence "represents an iterable sequence of elements. Unlike normal Iterators however, it doesn't expose
+      // a way for you to extract values from it, instead you provide it with a worker (an Accumulator) and that defines
+      // what happens with the data."
+      final Sequence<Cursor> cursors = adapter.makeCursors(
+          Filters.toFilter(null),
+          index.getDataInterval().withChronology(ISOChronology.getInstanceUTC()),
+          VirtualColumns.EMPTY,
+          Granularities.ALL,
+          false,
+          null
+      );
+      // Turn cursors into a Yielder
+      // A Yielder is an object like a linked list where the Yielder lets you access the current value with get()
+      // and lets you advance to the next item in the chain with next().
+      // A Yielder isDone() when there is nothing else in the chain.
+      // Also consider accumulating the Sequence instead of turning it into a Yielder
+      YieldingAccumulator<Cursor, Cursor> accumulator = new YieldingAccumulator<Cursor, Cursor>() {
+        @Override
+        public Cursor accumulate(Cursor accumulated, Cursor in) {
+          return in;
+        }
+      };
+      Yielder<Cursor> cursorYielder = cursors.toYielder(null, accumulator);
+      _cursor = cursorYielder.get();
+      // A Yielder must be closed to prevent resource leaks.
+      cursorYielder.close();
+
+      ColumnSelectorFactory columnSelectorFactory = _cursor.getColumnSelectorFactory();
+      _selectors = _columnNames
+          .stream()
+          .map(columnSelectorFactory::makeColumnValueSelector)
+          .collect(Collectors.toList());
+
+    } catch (Exception e) {
+    e.printStackTrace();
+    }
+  }
+
+  @Override
+  public void init(SegmentGeneratorConfig segmentGeneratorConfig)
+      throws IOException {
+    init(segmentGeneratorConfig.getInputFilePath(), segmentGeneratorConfig.getSchema());
+  }
+
+  @Override
+  public boolean hasNext() {
+    return !_cursor.isDone();
+  }
+
+  @Override
+  public GenericRow next() throws IOException {
+    return next(new GenericRow());
+  }
+
+  @Override
+  public GenericRow next(GenericRow reuse) throws IOException {
+    // TODO: use Pinot schema to fill the values to handle missing column and default values properly
+    // Only the columns whose names are in the Pinot schema will get processed
+    for (int i = 0; i < _columnNames.size(); i++) {
+      final String columnName = _columnNames.get(i);
+      final Object value = _selectors.get(i).getObject();
+      // Figure out how Pinot handles time; see TimeGranularitySpec and TimeFieldSpec
+      if (_columnNames.get(i).equals(ColumnHolder.TIME_COLUMN_NAME)) {
+        reuse.putField(columnName, new DateTime());
```

Comment:
Druid has a special time column `_time`. We need to appropriately handle the timestamp. By putting `new DateTime()`, you're overwriting the time column value. We should put the value what's read from Druid segment.

What the type of `Object value` for the time column returned by Druid segment?